### PR TITLE
nixos/picom: add experimentalBackends option

### PR DESCRIPTION
**Pull request by axelf4**
###### Motivation for this change

This option is only available as a command-line flag and not from the config file, that is `services.picom.settings`. Therefore it is more important that it gets its own option.

One reason one might need this set is that blur methods other than kernel do not work with the old backends, see yshui/picom#464.

For reference, the home-manager picom module exposes this option too.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/picom.nix
Comment (aanderse) [inline nixos/modules/services/x11/picom.nix:314]:
```suggestion
          + (optionalString cfg.experimentalBackends " --experimental-backends");
```

**Review by aanderse — APPROVED** [review]
Simple enough change, LGTM :+1: 

I've never used `picom` and have no specific knowledge of it, so it would be nice to get an approval from one of the maintainers before merge... ping @Twey @Enzime @ertes 

If we don't hear back in a few days feel free to ping me and we'll merge. Thanks!

**Comment by axelf4** [comment]
@aanderse It does not look like the maintainers are available.

**Comment by aanderse** [comment]
@axelf4 thanks for the PR and ping :tada:

**Pull request merged**
